source/internal/_isFunction.js

```diff
@@ -1,3 +1,7 @@
 export default function _isFunction(x) {
-  return Object.prototype.toString.call(x) === '[object Function]';
+  var type = Object.prototype.toString.call(x);
+  return type  === '[object Function]' ||
+    type === '[object AsyncFunction]' ||
+    type === '[object GeneratorFunction]' ||
+    type === '[object Proxy]';
```

Comment:
Is a proxy always callable?

Reply:
I'm actually not sure. I played around on the console in a `node` repl and I couldn't come to a conclusion. I personally don't use a lot of proxies and am not that familiar with edge cases. I would be fine removing the check for `'[object Proxy]'` from the list.

If someone feels strongly about keeping it, then perhaps the real answer is to put some guards in `R.invoker`?